librz/bin/bobj_process_class.c

```diff
@@ -0,0 +1,92 @@
+// SPDX-FileCopyrightText: 2023 RizinOrg <info@rizin.re>
+// SPDX-FileCopyrightText: 2023 deroad <wargio@libero.it>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+static void process_class_method(RzBinObject *o, RzBinSymbol *method) {
+	// rebase physical address
+	method->paddr += o->opts.loadaddr;
+
+	char *key = rz_str_newf(RZ_BIN_FMT_CLASS_HT_GLUE, method->classname, method->name);
+	if (!key) {
+		RZ_LOG_ERROR("bin: failed to allocate class-method string\n");
+		return;
+	}
+
+	if (!ht_pp_find(o->glue_to_class_method, key, NULL)) {
+		ht_pp_insert(o->glue_to_class_method, key, method);
+	}
+	free(key);
+
+	if (!ht_up_find(o->vaddr_to_class_method, method->vaddr, NULL)) {
+		ht_up_insert(o->vaddr_to_class_method, method->vaddr, method);
+	}
```

Comment:
you can just insert, it should ignore the insert if the key is already present.

Reply:
i guess this is ok to switch as you asked.